cpp/src/arrow/type_traits.h

```diff
@@ -1187,7 +1202,6 @@ constexpr bool is_var_length_list(Type::type type_id) {
   switch (type_id) {
     case Type::LIST:
     case Type::LARGE_LIST:
-    case Type::MAP:
```

Comment:
I don't think we want to change this?

Reply:
Well, 1) It is a public API, so we probably shouldn't change the current semantics, 2) one should review internal uses of it to make sure changing it doesn't break anything that wouldn't be adequately tested.